src/org/exist/xquery/functions/fn/JSON.java

```diff
@@ -144,10 +144,11 @@ private Sequence parseResource(Sequence href, String handleDuplicates, JsonFacto
             if (source == null) {
                 throw new XPathException(this, ErrorCodes.FOUT1170, "failed to load json doc from URI " + url);
             }
-            final InputStream is = source.getInputStream();
-            final JsonParser parser = factory.createParser(is);
-            final Item result = readValue(context, parser, handleDuplicates);
-            return result == null ? Sequence.EMPTY_SEQUENCE : result.toSequence();
+            try (final InputStream is = source.getInputStream()) {
+                final JsonParser parser = factory.createParser(is);
```

Comment:
Should we bring this one inside the `try( .. )` as well?

Reply:
Yes I think that is sensible. JsonParser#close() can free some internal buffers

Reply:
The same consideration should also be given to line 123 as well.